zkevm-circuits/src/evm_circuit/execution/push.rs

```diff
@@ -0,0 +1,284 @@
+use crate::{
+    evm_circuit::{
+        execution::{
+            bus_mapping_tmp::{Block, Call, ExecStep, Transaction},
+            ExecutionGadget,
+        },
+        step::ExecutionResult,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StateTransition, Transition::Delta,
+            },
+            sum, Cell, Word,
+        },
+    },
+    util::Expr,
+};
+use array_init::array_init;
+use bus_mapping::{eth_types::ToLittleEndian, evm::OpcodeId};
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct PushGadget<F> {
+    same_context: SameContextGadget<F>,
+    value: Word<F>,
+    selectors: [Cell<F>; 31],
```

Comment:
The first `selector` is removed since it's constraint to be always `1` previously.